src/app/Services/UserPostService.php

```diff
@@ -109,4 +110,25 @@ public function delete(int $userPostToDeleteId): void
     {
         UserPost::destroy($userPostToDeleteId);
     }
+
+    public function likePost(int $userPostToLikeId): void
+    {
+        PostLike::create([
+            'post_id' => $userPostToLikeId,
+            'user_id' => auth()->id()
+        ]);
+    }
+
+    public function unlikePost(int $userPostToUnlikeId): void
+    {
+        $postLike = PostLike::where('user_id', auth()->id())
+                    ->where('post_id', $userPostToUnlikeId)
+                    ->first()->id;
+        PostLike::destroy($postLike);
```

Comment:
It is advisable to use soft delete to preserve the data by not removing if from the db but only tagging it as deleted.

https://laravel.com/docs/11.x/eloquent#soft-deleting